src/sentry/grouping/api.py

```diff
@@ -101,6 +117,51 @@ def _get_enhancements(self, project) -> str:
         cache.set(cache_key, rv)
         return rv
 
+    def _get_fingerprinting(self, project: Project) -> dict[Any, Any]:
+        """
+        Returns the fingerprinting rules for a project.
+        Merges the project's custom fingerprinting rules (if any) with the default built-in rules.
+        """
+
+        config_id = self._get_config_id(project)
+
+        from sentry.grouping.fingerprinting import FingerprintingRules, InvalidFingerprintingConfig
+
+        try:
+            include_builtin = features.has(
+                "organizations:grouping-built-in-fingerprint-rules", project.organization
+            )
+        except Exception:
+            # workaround for project being mocked w/o organization is some tests
+            include_builtin = False
+
+        bases = CONFIGURATIONS[config_id].fingerprinting_bases if include_builtin else []
+
+        rules = project.get_option("sentry:fingerprinting_rules")
+
+        if not rules:
+            return FingerprintingRules([], bases=bases).to_json(include_builtin=include_builtin)
```

Comment:
This code path would not be taking advantage of the caching. That may be fine but I wanted to call it out if it's unintentional.

Reply:
The "caching" here is kinda useless anyway. The only thing it saves if _parsing_ from fingerprinting rules text DSL to config structure (JSON-able dict). In this case that config structure is empty.